native/src/main/java/io/ballerina/stdlib/http/api/service/signature/QueryParam.java

```diff
@@ -47,14 +48,26 @@
     }
 
     private void validateQueryParamType() {
-        if (isValidBasicType(typeTag) || (typeTag == TypeTags.ARRAY_TAG && isValidBasicType(
-                ((ArrayType) type).getElementType().getTag()))) {
+        if (isValidBasicType(typeTag) || isValidJsonMap() || (typeTag == TypeTags.ARRAY_TAG && (isValidBasicType(
```

Comment:
I think we can remove this `validateQueryParamType` validation as we have written the compiler plugin. Can you check removing this  whole method?